crypto/s2n_pkey.c

```diff
@@ -19,12 +19,17 @@
 
 #include "crypto/s2n_openssl_evp.h"
 #include "crypto/s2n_openssl_x509.h"
+#include "crypto/s2n_pkey_evp.h"
 #include "crypto/s2n_rsa_pss.h"
 #include "error/s2n_errno.h"
 #include "utils/s2n_mem.h"
 #include "utils/s2n_result.h"
 #include "utils/s2n_safety.h"
 
+#ifndef EVP_PKEY_RSA_PSS
+    #define EVP_PKEY_RSA_PSS EVP_PKEY_NONE
+#endif
```

Comment:
I moved this out of s2n_rsa_pss.h because rather alarmingly, EVP_PKEY_RSA_PSS is defined in openssl/evp.h and s2n_rsa_pss.h wasn't actually including openssl/evp.h. So EVP_PKEY_RSA_PSS only wasn't always EVP_PKEY_NONE because another included header happened to include openssl/evp.h.

This highlights the issue with #ifdefs in general, but in this particular case it's at least safer to do this in this file bc this file actually uses EVP_PKEY_RSA_PSS, so also requires other symbols from openssl/evp.h. We're less likely to just remove the openssl/evp.h include.